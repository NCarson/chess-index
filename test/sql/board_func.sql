

select expected_or_fail_bool(
    (select footer('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board)::text = 'w KQkq -'), true);

select expected_or_fail_bool(
    (select pcount('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board) = 32), true);

select expected_or_fail_bool(
    (select side('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board) = 'w'), true);

select expected_or_fail_bool(
    (select side('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR b KQkq - 0 1'::board) = 'b'), true);

select expected_or_fail_bool(
    (select pieceindex('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board, 'w') = 'QRRBBNNPPPPPPPP'), true);

select expected_or_fail_bool(
    (select pieceindex('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board, 'b') = 'QRRBBNNPPPPPPPP'), true);


select ('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board)::piecesquare[];

select remove_pieces('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board, 'p');

/*
select expected_or_fail_bool(
    (select pieces('p7/8/8/8/8/8/8/K7 w KQkq -'::board) = '{Kh8,ph1}'), true);
*/
