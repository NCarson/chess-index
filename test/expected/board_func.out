select expected_or_fail_bool(
    (select footer('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board)::text = 'w KQkq -'), true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(
    (select pcount('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board) = 32), true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(
    (select side('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board) = 'w'), true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(
    (select side('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR b KQkq - 0 1'::board) = 'b'), true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(
    (select pieceindex('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board, 'w') = 'QRRBBNNPPPPPPPP'), true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(
    (select pieceindex('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board, 'b') = 'QRRBBNNPPPPPPPP'), true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select pieces('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board);
                                                              pieces                                                               
-----------------------------------------------------------------------------------------------------------------------------------
 {ra8,nb8,bc8,qd8,ke8,bf8,ng8,rh8,pa7,pb7,pc7,pd7,pe7,pf7,pg7,ph7,Pa2,Pb2,Pc2,Pd2,Pe2,Pf2,Pg2,Ph2,Ra1,Nb1,Bc1,Qd1,Ke1,Bf1,Ng1,Rh1}
(1 row)

select ('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board)::piecesquare[];
                                                            piecesquare                                                            
-----------------------------------------------------------------------------------------------------------------------------------
 {ra8,nb8,bc8,qd8,ke8,bf8,ng8,rh8,pa7,pb7,pc7,pd7,pe7,pf7,pg7,ph7,Pa2,Pb2,Pc2,Pd2,Pe2,Pf2,Pg2,Ph2,Ra1,Nb1,Bc1,Qd1,Ke1,Bf1,Ng1,Rh1}
(1 row)

select remove_pieces('rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board, 'p');
             remove_pieces              
----------------------------------------
 rnbqkbnr/8/8/8/8/8/8/RNBQKBNR w KQkq -
(1 row)

